### PR TITLE
[Feature:TAGrading] websockets in grading inquiry

### PR DESCRIPTION
**Pull request by marwanatef2**
### What is the current behavior?
Students/Graders should refresh the page to get new posts in the grade inquiry discussion which may be boring if the discussion goes a long way or may be they will forget and the inquiry will not be resolved in the same time of discussion.

### What is the new behavior?
The grade inquiry discussion is updated on the go with websockets whether new posts are added of the inquiry is resolved users will get the updates immediately.

### Other information?
In order to test you'll need to have websockets enabled, so please check the [docs](https://submitty.org/sysadmin/version_notes/v20.06.00)

### CODE REVIEW

File: site/public/js/grade-inquiry.js
Comment (mukul-kmr-jha) [inline site/public/js/grade-inquiry.js:144]: Can you add curly braces for the `if` condition even if its single liner condition because its more readable

File: site/app/libraries/database/DatabaseQueries.php
Comment (mukul-kmr-jha) [inline site/app/libraries/database/DatabaseQueries.php:4069]: can you remove the var `regrade_post_id` if its not used in the function?

File: site/app/controllers/student/GradeInquiryController.php
Comment (mukul-kmr-jha) [inline site/app/controllers/student/GradeInquiryController.php:213]: can you remove the var `regrade_post_id` if it's not used in the function?

File: site/app/libraries/database/DatabaseQueries.php
Comment (marwanatef2) [reply site/app/libraries/database/DatabaseQueries.php:4069]: `insertNewRegradePost` returns the _post_id_ that's why i store it in a variable even if it's not used

File: site/app/controllers/student/GradeInquiryController.php
Comment (marwanatef2) [reply site/app/controllers/student/GradeInquiryController.php:213]: please refer to above discussion

**Comment by marwanatef2** [comment]
@mukul-kmr-jha  Thank you for your review,

> When I add a post it is received as a pair of posts to the user listening for this at the other end.

Can you tell me when do you get this issue so I can reproduce and hopefully fix.

> All the grade inquiry conversations are public and is visible to any third person in between

Will fix this after merging master, we used to store socket clients per course not page

**Comment by marwanatef2** [comment]
Also merging master will produce unwanted/unexpected behavior because the socket configuration was updated on another PR that was merged after I've started this PR, so I'm waiting for #5794 to get merged so I can fix this. @mukul-kmr-jha

File: site/app/libraries/database/DatabaseQueries.php
Comment (mukul-kmr-jha) [reply site/app/libraries/database/DatabaseQueries.php:4069]: I dont think you need to store the post_id just because its returned by the called function... let's remove this one... thanks:)

**Comment by mukul-kmr-jha** [comment]
> Can you tell me when do you get this issue so I can reproduce and hopefully fix. 

I enabled grade inquiry on one of the gradeable and started posting grade_inquiry and when I add post from student view I received two posts on instructor side and same things happen when I reply from instructor side... and to remove this duplicates posts I needed a page refresh

**Comment by marwanatef2** [comment]
Hi @mukul-kmr-jha , please pull latest changes, run `submitty_install clean` in vagrant , and retest, you should not see any errors.
Thanks.

**Comment by marwanatef2** [comment]
@mukul-kmr-jha Thanks for this helpful review, will address those changes and get back to you in a day or something.

**Comment by shailpatels** [comment]
@marwanatef2 please address @mukul-kmr-jha 's code review.

To reiterate what he said if you're not using the return value of a function you can safely disregard it 
https://stackoverflow.com/q/3709420

You should also use curly braces in JS code, one line if statements are less readable and are easier to run into bugs

**Review by mukul-kmr-jha — APPROVED** [review]
Works really good and does all the mentioned things:)

**Pull request merged**
